src/extension.ts

```diff
@@ -249,6 +259,53 @@ export async function attach(
     } while (useRetry && Date.now() - startTime < timeout);
 }
 
+export async function attachToCurrentDebugTarget(context: vscode.ExtensionContext): Promise<void> {
+    if (!telemetryReporter) {
+        telemetryReporter = createTelemetryReporter(context);
+    }
+
+    telemetryReporter.sendTelemetryEvent('command/attachToCurrentDebugTarget');
+
+    // Attempt to attach to active CDP target
+    const session = vscode.debug.activeDebugSession;
+    if (!session) {
+        const errorMessage = 'No active debug session';
+        telemetryReporter.sendTelemetryErrorEvent('command/attachToCurrentDebugTarget/devtools', {message: errorMessage});
+        void vscode.window.showErrorMessage(errorMessage);
+        return;
+    }
+
+    let targetWebsocketUrl;
+    try {
+        const addr: RequestCDPProxyResult|undefined = await vscode.commands.executeCommand(
+        'extension.js-debug.requestCDPProxy',
+        session.id,
+        );
+        if (addr) {
+            const formed = `ws://${addr.host}:${addr.port}${addr.path || ''}`;
+            targetWebsocketUrl = formed;
+        }
+    } catch (e) {
+        if (e instanceof Error) {
```

Comment:
why is this needed? any other exception will be swallowed by the return statement.

Reply:
Typescipt gets mad if you access e.message without first checking if it's an Error instead of Any

Reply:
got it, that makes sense. My only suggestion would be to rethrow to raise unexpected exceptions.
`if (e instanceof Error) {
            telemetryReporter.sendTelemetryErrorEvent('command/attachToCurrentDebugTarget/devtools', {message: e.message});
            void vscode.window.showErrorMessage(e.message);
        } else{
            throw e;
        }`